fix: ensure requestTimeout and maxRetries are integers

tgarciaalv opened the pull request:
This is failing when environment variables are used for configuration. Env vars are strings but are never converted to int before using them in arithmetic operations.

My configuration:
 ```
               "OKTA_CLIENT_CONNECTIONTIMEOUT": "30",
                "OKTA_CLIENT_ORGURL": "https://dev-89428916.okta.com",
                "OKTA_CLIENT_AUTHORIZATIONMODE": "PrivateKey",
                "OKTA_CLIENT_CLIENTID": "${env:OKTA_CLIENT_CLIENTID}",
                "OKTA_CLIENT_SCOPES": "okta.users.read,okta.groups.read",
                "OKTA_CLIENT_PRIVATEKEY": "${env:OKTA_CLIENT_PRIVATEKEY}",
                "OKTA_CLIENT_REQUESTTIMEOUT": "0",
                "OKTA_CLIENT_RATELIMIT_MAXRETRIES": "4",
                "OKTA_CLIENT_LOGGING_ENABLED": "true",
                "OKTA_CLIENT_LOGGING_LEVEL": "INFO",
```

My error:
```
        # Raise Value Error if numerical inputs are invalid (< 0)
        self._request_timeout = config["client"].get('requestTimeout', 0)
>       if self._request_timeout < 0:
E       TypeError: '<' not supported between instances of 'str' and 'int'

/opt/homebrew/lib/python3.11/site-packages/okta/request_executor.py:35: TypeError
```

The os.environ dictionary in Python expects its values to be strings.

tgarciaalv committed:
fix: ensure requestTimeout and maxRetries are integers

This is failing when environment variables are used for configuration. Env vars are strings but are never converted to int before using them in arithmetic operations.

        # Raise Value Error if numerical inputs are invalid (< 0)
        self._request_timeout = config["client"].get('requestTimeout', 0)
>       if self._request_timeout < 0:
E       TypeError: '<' not supported between instances of 'str' and 'int'

/opt/homebrew/lib/python3.11/site-packages/okta/request_executor.py:35: TypeError

bryanapellanes-okta commented:
@tgarciaalv I apologize for the delayed response and thank you for your submission.  To ensure integrity, please provide a unit test that fails prior to your proposed change and passes with your change in place.  

We appreciate your engagement and thanks for using Okta!

tgarciaalv committed:
test: add unit tests for string-to-int conversion of requestTimeout and maxRetries

tgarciaalv committed:
fix: replace flatdict with flatdict2 for Python 3.13 compatibility

flatdict==4.0.1 (last release 2020) uses pkg_resources in its setup.py,
which was removed from Python 3.13 stdlib. flatdict2 is a maintained
fork with identical API and pre-built wheels.
